### DISCUSSION TITLE
Real robot integration using qibullet

### DISCUSSION THREAD

**Discussion by hebbbbaa** [softbankrobotics-research/qibullet#87] [Q&A]
I want to integrate a robot (Pepper by SoftBank) with qibullet to output what I have trained the virtual robot to do. is there a way (piece of code) to help me connect the robot to qibullet? (via physical IP for example)

**Comment by mbusy** [comment]
The simulation API has been designed to be as transparent as possible with the NAOqi API, using the NAOqi API doc available on the Internet you should be able to find a simple piece of Python code controlling the joints of the robot (using ALmotion). Adapting the code you have for a virtual robot shouldn't take too long (it depends on the version of NAOqi that you are running though, with 2.9 this comment is not accurate)
